### ISSUE TITLE
High loss in LSTM model with Keras

### ISSUE THREAD

**Issue by abhinavme** [abhinavme/CI-CD-Test#1]
I am trying to build a deep learning chatbot using Encoder Decoder model with LSTM network in Keras. I am using the cornell movie dialogs corpus. This is the model -

```
model = keras.Sequential()

encoder_input = layers.Input(shape=(None,))
encoder_embedded = layers.Embedding(input_dim=num_encoder_tokens, output_dim=64, input_length=25)(
    encoder_input
)

# Return states in addition to output
output, state_h, state_c = layers.LSTM(64, return_state=True, dropout=0.2, name="encoder")(
    encoder_embedded
)
encoder_state = [state_h, state_c]

decoder_input = layers.Input(shape=(None,))
decoder_embedded = layers.Embedding(input_dim=num_decoder_tokens, output_dim=64, input_length=30)(
    decoder_input
)

# Pass the 2 encoder states to a new decoder LSTM layer as initial state

decoder_lstm = layers.LSTM(64, dropout=0.2, return_state=True)
decoder_output, _, _ = decoder_lstm(decoder_embedded, initial_state=encoder_state)

decoder_dense = layers.Dense(max_length_answers, activation='softmax')
decoder_output = decoder_dense(decoder_output)  

model = keras.Model([encoder_input, decoder_input], decoder_output)
#model.summary()

opt_adam = keras.optimizers.Adam(learning_rate = 0.001)

model.compile(optimizer=opt_adam,loss='mean_absolute_error',metrics=["accuracy"])

tensorboard = TensorBoard(log_dir="logs/{}".format(time()))

model.fit([padded_questions, testing_answers],decoder_target,
          batch_size=64,
          epochs=3,
          shuffle=True,
          validation_split = 0.15,
          callbacks = [tensorboard]
         )
```

My problem is that I am getting very high loss specially when I used mean_squared_error as the loss function -

```
Epoch 1/3
2709/2709 [==============================] - 222s 81ms/step - loss: 62964215.6118 - accuracy: 0.8577 - val_loss: 62386828.0000 - val_accuracy: 1.0000
Epoch 2/3
2709/2709 [==============================] - 223s 82ms/step - loss: 63013080.8369 - accuracy: 1.0000 - val_loss: 62386828.0000 - val_accuracy: 1.0000
Epoch 3/3
2596/2709 [===========================>..] - ETA: 8s - loss: 62882140.3713 - accuracy: 1.0000
```

I even tried changing the loss function to mean_absolute_error. Still loss is high -

```
Epoch 1/3
2709/2709 [==============================] - 218s 79ms/step - loss: 7221.6704 - accuracy: 0.4116 - val_loss: 7161.3008 - val_accuracy: 1.0000
Epoch 2/3
2709/2709 [==============================] - 223s 82ms/step - loss: 7225.7182 - accuracy: 1.0000 - val_loss: 7161.3008 - val_accuracy: 1.0000
Epoch 3/3
1851/2709 [===================>..........] - ETA: 1:06 - loss: 7220.1683 - accuracy: 1.0000
```

I even tried with different learning rates, but the loss doesn't come down. Also, the Validation Accuracy reaches 100% at 1st epoch only and doesn't change after that, Is it normal or is there something wrong with the model architecture ?
 Any help would be appreciated.
